versioned/persist/tx/src/main/java/org/projectnessie/versioned/persist/tx/TxDatabaseAdapter.java

```diff
@@ -281,6 +299,20 @@ public Hash transplant(
                     h -> {},
                     updateCommitMetadata);
 
+            Hash parentId = getRefLogHead(conn);
```

Comment:
This whole new code block looks very similar to the other usages. Can it be moved to a separate method? (Same for the NonTxDatabaseAdapter maybe)